### ISSUE TITLE
The "Intercept requests(#28)" is a glitch for Facebook login

### ISSUE THREAD

**Issue by hykw** [ueberauth/ueberauth#44]
As of String.replace_trailing in "Intercept requests with and without a trailing slash(#28)", Facebook login doesn't work correctly.

Due to the replacing, `conn`, which will be passed to the callbacks(like following) don't include ueberauth_auth in assigns at all.

```
def facebook_callback(%{ assigns: %{ ueberauth_failure: _fails } } = conn, params) do  
```

It works correctly without the code in [lib/ueberauth.ex](https://github.com/ueberauth/ueberauth/commit/ca31f6f711060deaf72c1712bd1e9f4cb454dae6#diff-3f81f8ab5bb0951f0fa5db7604b4b9b3R205), like below:

```
#if strategy = Map.get(opts, String.replace_trailing(request_path, "/", "")) do
if strategy = Map.get(opts, request_path) do
```

**Comment by hassox** [comment]
@hykw Is this still an issue for you?

**Comment by hykw** [comment]
@hassox 
Yes, it is. I've updated to ueberauth 0.4.0/ueberauth_facebook 0.6.0 and tried it. It still occurs the same issue. So, the workaround is still to change the line as I mentioned(`if strategy...` to `if strategy = Map.get(opts, request_path) do`).
